### ISSUE TITLE
-Wlogical-not-parentheses in drivers/scsi/qla2xxx/qla_nx.c

### ISSUE THREAD

**Issue by shenki** [ClangBuiltLinux/linux#80]
```c
  CC [M]  drivers/scsi/qla2xxx/qla_nx.o
drivers/scsi/qla2xxx/qla_nx.c:3702:7: warning: logical not is only applied to the left hand side of this
      comparison [-Wlogical-not-parentheses]
                if (!qla2x00_eh_wait_for_pending_commands(vha, 0, 0,
                    ^
drivers/scsi/qla2xxx/qla_nx.c:3702:7: note: add parentheses after the '!' to evaluate the comparison first
                if (!qla2x00_eh_wait_for_pending_commands(vha, 0, 0,
                    ^
                     (
drivers/scsi/qla2xxx/qla_nx.c:3702:7: note: add parentheses around left hand side expression to silence
      this warning
                if (!qla2x00_eh_wait_for_pending_commands(vha, 0, 0,
                    ^
                    (
```

First glance suggests changing the test to `if (qla2x00_eh_wait_for_pending_commands(...) != QUA_SUCESSS)`. The more you look the stranger the code becomes.

**Comment by nathanchance** [comment]
`if (!qla2x00_eh_wait_for_pending_commands(vha, 0, 0, WAIT_HOST) == QLA_SUCCESS)`

`if (!qla2x00_eh_wait_for_pending_commands(vha, 0, 0, WAIT_HOST) == 0)`

`if (!!qla2x00_eh_wait_for_pending_commands(vha, 0, 0, WAIT_HOST))`

`if (qla2x00_eh_wait_for_pending_commands(vha, 0, 0, WAIT_HOST))`

is probably the proper simplification.

**Comment by nathanchance** [comment]
Forgot to link patch submitted: https://lore.kernel.org/lkml/20180920201002.23979-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=0bfe7d3cae58f02a05c39fba6829d6e7b50802f5

**Comment by nickdesaulniers** [comment]
@nathanchance mentions that this was accidentally reverted by commit 3695310e37b4 ("scsi: qla2xxx: Update flash read/write routine") and has sent a fix. https://lore.kernel.org/r/20190320163315.12740-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
I resent the fix with a more condensed commit message and cc'd the authors who reintroduced the warning: https://lore.kernel.org/lkml/20190326220444.10040-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
Second fix is accepted, I'll close this again when it shows up in -next: https://git.kernel.org/mkp/scsi/c/46333cebe77883ff38be59c3e7200c5a5145d6ee

**Comment by nathanchance** [comment]
Fixed in `next-20190329`: https://git.kernel.org/next/linux-next/c/46333cebe77883ff38be59c3e7200c5a5145d6ee

Hopefully it stays fixed now :)
